lib/http/HttpClient_WinInet.cpp

```diff
@@ -325,11 +325,7 @@ class WinInetRequestWrapper
             // trigger INTERNET_STATUS_REQUEST_COMPLETE again.
 
             m_bodyBuffer.insert(m_bodyBuffer.end(), m_buffer, m_buffer + m_bufferUsed);
-            do {
-                if (m_responsePending && m_bufferUsed == 0) {
-                    LOG_TRACE("InternetReadFile finished async");
-                    break;
-                }
+            while (!m_responsePending || m_bufferUsed != 0) {
```

Comment:
Need to make sure "m_bufferUsed != 0" is re-evaluated after IRF is called (not after IRF returns pending). So perhaps rename this to something like m_readingData instead?